### PR TITLE
Feature/id ripper speedup2

### PR DESCRIPTION
**Pull request by timosachsenberg**
## Description

significant speed-up from hours to seconds

## Checklist
- [ ] Make sure that you are listed in the AUTHORS file
- [ ] Add relevant changes and new features to the CHANGELOG file
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] New and existing unit tests pass locally with my changes
- [ ] Updated or added python bindings for changed or new classes (Tick if no updates were necessary.)

### How can I get additional information on failed tests during CI
<details>
  <summary>Click to expand</summary>
If your PR is failing you can check out

- The details of the action statuses at the end of the PR or the "Checks" tab.
- http://cdash.openms.de/index.php?project=OpenMS and look for your PR. Use the "Show filters" capability on the top right to search for your PR number.
  If you click in the column that lists the failed tests you will get detailed error messages.

</details>

### Advanced commands (admins / reviewer only)
<details>
  <summary>Click to expand</summary>
  
- `/reformat` (experimental) applies the clang-format style changes as additional commit. Note: your branch must have a different name (e.g., yourrepo:feature/XYZ) than the receiving branch (e.g., OpenMS:develop). Otherwise, reformat fails to push.
- setting the label "NoJenkins" will skip tests for this PR on jenkins (saves resources e.g., on edits that do not affect tests)
- commenting with `rebuild jenkins` will retrigger Jenkins-based CI builds
  
</details>

---
:warning: Note: Once you opened a PR try to minimize the number of *pushes* to it as every push will trigger CI (automated builds and test) and is rather heavy on our infrastructure (e.g., if several pushes per day are performed).

### CODE REVIEW

**Comment by timosachsenberg** [comment]
tests pass and results look good to me.
approx. 1000x speedup. now as fast as merging on the data I tested

File: src/openms/source/ANALYSIS/ID/IDRipper.cpp [hunk 1]
```diff
@@ -219,70 +225,124 @@ namespace OpenMS
...
+          for (const ProteinHit& prot : proteins_of_accessions)
+          {              
+            // check if protein has already been added
+            const String& acc = prot.getAccession();
+             auto& acc_set = ripped_prot_map[rfi][merged_prot_identifier];
+            if (auto ri = acc_set.find(acc); ri == acc_set.end())
+            { // only add protein once to the run identifier
+              p.insertHit(prot);;
+              acc_set.insert(acc);
+              //std::cout << "ripped/merged identifier: " << ripped_prot_identifier << " " << prot << std::endl;
```
Comment (poshul) [inline src/openms/source/ANALYSIS/ID/IDRipper.cpp:334]: Delete

**Review by poshul — APPROVED** [review]

**Review by jpfeuffer — CHANGES_REQUESTED** [review]
Please add another fix

File: src/openms/source/ANALYSIS/ID/IDRipper.cpp [hunk 2]
```diff
@@ -219,70 +225,124 @@ namespace OpenMS
...
+      if (RipFileMap::iterator it = ripped.find(rfi); it == ripped.end())
+      { // file identifier does not exist yet. We need to create it.
+        OPENMS_LOG_INFO << "Creating entry for file identifier:\n" 
+                        << "File origin: " << rfi.getOriginFullname() << "\n"
+                        // << "Identification run index: " << rfi.getIdentRunIdx() << "\n" // not set here?
+                        << "Basename: " << rfi.getOutputBasename() << "\n"
+                        << "Merged identification file run identifier: " << merged_prot_identifier << "\n"
+                        << std::endl;
+        // create the protein run but only set the protein hits that are needed for the current peptide identification
+        ProteinIdentification p{merged_protein_id_run};
```
[the remark this answers is not in the document]
Comment (jpfeuffer) [reply src/openms/source/ANALYSIS/ID/IDRipper.cpp:272]: @timosachsenberg

[the remark this answers is not in the document]
Comment (timosachsenberg) [reply src/openms/source/ANALYSIS/ID/IDRipper.cpp:272]: Thanks. Yeah I noticed these lines are inefficient and forgot to change it. I can look at it - sounds like ab easy fix.

[the remark this answers is not in the document]
Comment (jpfeuffer) [reply src/openms/source/ANALYSIS/ID/IDRipper.cpp:272]: Yes if every protein comes from a different file.
Just write a custom function in the proteinID copyMetadataOnly. It doesn't have so many members.

**Pull request merged**